File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/GravityBooster.java
package org.rexcellentgames.burningknight.entity.item.accessory.equipable;

import org.rexcellentgames.burningknight.assets.Locale;
import org.rexcellentgames.burningknight.entity.item.pet.impl.Orbital;
import org.rexcellentgames.burningknight.util.Tween;

public class GravityBooster extends Equipable {
	{
		name = Locale.get("gravity_booster");
		description = Locale.get("gravity_booster_desc");
		sprite = "item-blank_card";
	}

	@Override
	public void onEquip(boolean load) {
		super.onEquip(load);

		Tween.to(new Tween.Task(Orbital.speed + 1, 0.3f) {
			@Override
			public float getValue() {
				return Orbital.speed;
			}

			@Override
			public void setValue(float value) {
				Orbital.speed = value;
			}
		});
	}

	@Override
	public void onUnequip(boolean load) {
		super.onUnequip(load);

		Tween.to(new Tween.Task(Orbital.speed - 1, 0.3f) {
			@Override
			public float getValue() {
				return Orbital.speed;
			}

			@Override
			public void setValue(float value) {
				Orbital.speed = value;
			}
		});
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/GravityBooster.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equipable

import org.rexcellentgames.burningknight.assets.Locale
import org.rexcellentgames.burningknight.entity.item.pet.impl.Orbital
import org.rexcellentgames.burningknight.util.Tween

class GravityBooster : Equipable() {
	init {
		name = Locale.get("gravity_booster")
		description = Locale.get("gravity_booster_desc")
		sprite = "item-blank_card"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)

		Tween.to(object : Tween.Task(Orbital.speed + this.level, 0.3f) {
			override fun getValue(): Float {
				return Orbital.speed
			}

			override fun setValue(value: Float) {
				Orbital.speed = value
			}
		})
	}

	override fun upgrade() {
		this.level++;
		this.onEquip(false);
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)

		Tween.to(object : Tween.Task(Orbital.speed - this.level, 0.3f) {
			override fun getValue(): Float {
				return Orbital.speed
			}

			override fun setValue(value: Float) {
				Orbital.speed = value
			}
		})
	}
}